model/task_lifecycle.go

```diff
@@ -727,21 +676,47 @@ func RestartItemsAfterVersion(cq *commitqueue.CommitQueue, project, version, cal
 	return catcher.Resolve()
 }
 
-func tryDequeueAndAbortCommitQueueVersion(t *task.Task, cq commitqueue.CommitQueue, caller string) error {
-	p, err := patch.FindOne(patch.ByVersion(t.Version))
+// DequeueAndRestartForTask restarts all items after the given task's version, aborts/dequeues the current version,
+// and sends an updated status to GitHub.
+func DequeueAndRestartForTask(cq *commitqueue.CommitQueue, t *task.Task, githubState message.GithubState, caller, reason string) error {
+	if cq == nil {
+		var err error
+		cq, err = commitqueue.FindOneId(t.Project)
+		if err != nil {
+			return errors.Wrapf(err, "can't get commit queue for id '%s'", t.Project)
+		}
+		if cq == nil {
+			return errors.Errorf("no commit queue found for '%s'", t.Project)
+		}
+	}
+	// this must be done before dequeuing so that we know which entries to restart
+	if err := RestartItemsAfterVersion(cq, t.Project, t.Version, caller); err != nil {
+		return errors.Wrap(err, "unable to restart versions")
```

Comment:
Should we maybe add more details to this error?

Reply:
I'm thinking the user may not even know why anything is being restarted, and this error is not wrapped.